lib/pbench/test/unit/server/query_apis/test_iteration_samples.py

```diff
@@ -263,10 +261,14 @@ def test_query(
         }
         index = self.build_index_from_metadata()
 
-        if HeaderTypes.is_valid(build_auth_header["header_param"]):
-            expected_status = HTTPStatus.OK
-        else:
-            expected_status = HTTPStatus.FORBIDDEN
+        # get_expected_status() expects to read username and access from the
+        # JSON client payload, however this API acquires that information
+        # from the Dataset. Construct a fake payload corresponding to the
+        # attach_dataset fixture.
+        auth_json = {"user": "drb", "access": "private"}
```

Comment:
It's a bit more complicated than that since I'd have to look up the appropriate Dataset anyway from the set created by the fixture, which means I still need to assume at least the user. It might have been a bit more intellectually satisfying, but our tests make an enormous amount of assumptions about the users and datasets created by the fixtures and "generalizing" here just didn't seem worth the effort.